### PR TITLE
Fix multiple input-related problems on SDL

### PR DESCRIPTION
**Pull request by ThomasMiz**
# Summary of the PR
This PR fixes input events not being registered correctly and the window hanging problem from issue #698.

### CODE REVIEW

**Comment by ThomasMiz** [comment]
I think on line 96 of sdlView.cs I should add a '?' after System.Action. Kinda odd how nullable interacts with events huh

File: src/Input/Silk.NET.Input.Sdl/SdlKeyboard.cs
Comment (Perksey) [inline src/Input/Silk.NET.Input.Sdl/SdlKeyboard.cs:70]: nit: use braces

```suggestion
                        for (int i = 0; i < 32 && chars[i] != '\0'; i++)
                        {
                            KeyChar.Invoke(this, chars[i]);
                        }
```

File: src/Input/Silk.NET.Input.Sdl/SdlKeyboard.cs
Comment (Perksey) [inline src/Input/Silk.NET.Input.Sdl/SdlKeyboard.cs:63]: nit: prefer identity comparisons

```suggestion
                    if (KeyChar is not null)
```

**Review by Perksey — APPROVED** [review]
Few nitpicks for you to look at before you hit the merge button, but I'm happy with this.

File: src/Windowing/Silk.NET.Windowing.Sdl/SdlView.cs
Comment (ThomasMiz) [inline src/Windowing/Silk.NET.Windowing.Sdl/SdlView.cs:96]:
```suggestion
        public event System.Action? ProcessingEvents;
```

**Pull request merged**
